res/controllers/Traktor-Kontrol-S2-MK1-hid-scripts.js

```diff
@@ -0,0 +1,1361 @@
+        throw new Error("ButtonBrightnessOn must be a number between 0 and 0x1f (31).\n" +
+            "ButtonBrightnessOn is: " + ButtonBrightnessOn);
+    }
+    if (ButtonBrightnessOn < ButtonBrightnessOff) {
+        throw new Error("ButtonBrightnessOn must be greater than ButtonBrightnessOff.\n" +
+            "ButtonBrightnessOn is: " + ButtonBrightnessOn + "\n" +
+            "ButtonBrightnessOff is: " + ButtonBrightnessOff);
+    }
+
+    TraktorS2MK1.registerInputPackets();
+
+    var debugLEDs = false;
+    if (debugLEDs) {
+        var data = [0x1f, 0x1f, 0x1f, 0x1f, 0x1f, 0x1f, 0x1f, 0x1f, 0x1f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x00, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f, 0x7f];
+        controller.send(data, data.length, 0x80);
+    } else {
+        TraktorS2MK1.registerOutputPackets();
+    }
+
+    TraktorS2MK1.controller.setOutput("[Master]", "!warninglight", 0x00, true);
+    TraktorS2MK1.lightDeck("[Channel1]");
+    TraktorS2MK1.lightDeck("[Channel2]");
+    TraktorS2MK1.lightDeck("[EffectRack1_EffectUnit1]");
+    TraktorS2MK1.lightDeck("[EffectRack1_EffectUnit2]");
+};
+
+TraktorS2MK1.shutdown = function() {
+    var data = [];
+    for (var i = 0; i < 61; i++) {
+        data[i] = 0;
+    }
+    controller.send(data, data.length, 0x80);
+};
+
+TraktorS2MK1.incomingData = function(data, length) {
+    TraktorS2MK1.controller.parsePacket(data, length);
+};
+
+// The short message handles buttons and jog wheels.
+TraktorS2MK1.shortMessageCallback = function(packet, data) {
+    for (var name in data) {
+        var field = data[name];
+        if (field.name === "!jog_wheel") {
+            TraktorS2MK1.controller.processControl(field);
+            continue;
+        }
+
+        TraktorS2MK1.controller.processButton(field);
+    }
+};
+
+// There are no buttons handled by the long message, so this is a little simpler.
+TraktorS2MK1.longMessageCallback = function(packet, data) {
+    for (var name in data) {
+        var field = data[name];
+        TraktorS2MK1.controller.processControl(field);
+    }
+};
+
+TraktorS2MK1.samplerGainKnob = function(field) {
+    for (var i = 1; i <= 8; i++) {
+        engine.setParameter("[Sampler" + i + "]", "pregain", field.value / 4096);
+    }
+};
+
+TraktorS2MK1.toggleButton = function(field) {
+    if (field.value > 0) {
+        script.toggleControl(field.group, field.name);
+    }
+};
+
+TraktorS2MK1.shift = function(field) {
+    var shiftPressed = field.value > 0;
+    TraktorS2MK1.shiftPressed[field.group] = shiftPressed;
+    TraktorS2MK1.controller.setOutput(field.group, "!shift",
+        shiftPressed ? ButtonBrightnessOn : ButtonBrightnessOff, field.group, "!shift",
+        !TraktorS2MK1.batchingLEDUpdate);
+};
+
+TraktorS2MK1.loadTrackButton = function(field) {
+    if (TraktorS2MK1.shiftPressed[field.group]) {
+        engine.setValue(field.group, "eject", field.value);
```

Comment:
The argument must be the number of the deck to clone -> always the other deck.

Reply:
If i press `Shift + Load A` it clones deck B to A. The other way around does also work.

Reply:
Is field.value maybe 0? Does it clone when you press the button, or when you release it?
In case f 0 it should clone the first playing deck: https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-CloneFromDeck